R/config_spark.R

```diff
@@ -2,24 +2,33 @@
 #'
 #' @export
 #' @param file Name of the configuration file
-#' @param use_default TRUE to use the built-in detaults provided in this package
+#' @param use_default TRUE to use the built-in defaults provided in this package for local deplyomnet - \href{https://github.com/rstudio/sparklyr/blob/master/inst/conf/config-template.yml}{see}
+#' @param yarn TRUE to use the built-in defaults provided in this package for yarn cluster deplyomnet - \href{https://github.com/rstudio/sparklyr/blob/master/inst/conf/config-template.yml}{see}
 #'
 #' @details
 #'
 #' Read Spark configuration using the \pkg{\link[config]{config}} package.
 #'
 #' @return Named list with configuration data
-spark_config <- function(file = "config.yml", use_default = TRUE) {
+spark_config <- function(file = "config.yml", use_default = TRUE, yarn = FALSE) {
```

Comment:
As `spark_config` is used in `spark_connect` where there exists `master` parameter, maybe it could be

````{R}
spark_config <- function(file = "config.yml", use_default = TRUE, yarn = grepl("yarn", master))
````

for user interface simplification?